app/controllers/concerns/date_range_parser.rb

```diff
@@ -0,0 +1,48 @@
+require 'active_support/concern'
+
+module Concerns::DateRangeParser
+  extend ActiveSupport::Concern
+
+  included do
+    def parse_date_range(date_range)
+      date_to_range(date_range)
+    end
+  end
+
+private
+
+  def date_to_range(date_range)
+    case date_range
+    when 'last-month'
+      {
+        from: Time.zone.today.last_month.beginning_of_month.to_s,
+        to: Time.zone.today.last_month.end_of_month.to_s
+      }
+    when 'last-3-months'
+      {
+        from: (Time.zone.today - 3.months).to_s,
+        to: Time.zone.today.to_s
+      }
+    when 'last-6-months'
+      {
+        from: (Time.zone.today - 6.months).to_s,
+        to: Time.zone.today.to_s
+      }
+    when 'last-1-year'
+      {
+        from: (Time.zone.today - 1.year).to_s,
+        to: Time.zone.today.to_s
+      }
+    when 'last-2-years'
+      {
+        from: (Time.zone.today - 2.years).to_s,
+        to: Time.zone.today.to_s
+      }
+    else
```

Comment:
Not sure about the `else` statement. If the parameter value is not one of the expected ones, shouldn't we just fail?

Reply:
When we first render a single item page I believe we are defaulting to 30 days i.e before anyone has had the chance to pick a date range

Reply:
If that is the case, shouldn't that option be the default checkbox selection? So, the filter is always consistent with the result of the list? This might be something to discuss about with Jeremy/Polly.